src/button/Button.tsx

```diff
@@ -6,115 +6,124 @@ import Spinner from "../spinner/Spinner";
 // SCSS import is moved here to be able to override spinner styles without nesting
 import "./_button.scss";
 
-export interface ButtonProps {
-  testid?: string;
+export type ButtonProps = Omit<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  "disabled"
+> & {
   children: React.ReactNode;
+  testid?: string;
   customSpinner?: React.ReactNode;
+  isDisabled?: boolean;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  shouldFocus?: boolean;
   onClick?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseOver?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseDown?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseUp?: React.ReactEventHandler<HTMLButtonElement>;
   onFocus?: React.ReactEventHandler<HTMLButtonElement>;
   onBlur?: React.ReactEventHandler<HTMLButtonElement>;
   id?: string;
+  ref?: React.RefObject<HTMLButtonElement>;
   lang?: string;
-  type?: "button" | "submit" | "reset";
   shouldDisplaySpinner?: boolean;
-  isDisabled?: boolean;
-  shouldPreventDefault?: boolean;
-  shouldStopPropagation?: boolean;
-  shouldFocus?: boolean;
   ariaLabel?: string;
   customClassName?: string;
   tabIndex?: number;
```

Comment:
Please clean out the props that are already available from `React.ButtonHTMLAttributes<HTMLButtonElement>`. This was mentioned in my comment that proposed using `React.ButtonHTMLAttributes<HTMLButtonElement>`